addons/test_website/tests/test_redirect.py

```diff
@@ -74,88 +72,88 @@ def test_02_redirect_308_RequestUID(self):
         WebsiteHttp = odoo.addons.website.models.ir_http.Http
 
         def _get_error_html(env, code, value):
-            return str(code).split('_')[-1], "CUSTOM %s" % code
+            return str(code).split('_')[-1], f"CUSTOM {code}"
 
         with patch.object(WebsiteHttp, '_get_error_html', _get_error_html):
             # Patch will avoid to display real 404 page and regenerate assets each time and unlink old one.
             # And it allow to be sur that exception id handled by handle_exception and return a "managed error" page.
 
             # published
-            resp = self.url_open("/test_website/200/name-%s" % rec_published.id, allow_redirects=False)
+            resp = self.url_open(f"/test_website/200/name-{rec_published.id}", allow_redirects=False)
             self.assertEqual(resp.status_code, 308)
-            self.assertEqual(resp.headers.get('Location'), self.base_url + "/test_website/308/name-%s" % rec_published.id)
+            self.assertEqual(resp.headers.get('Location'), f"{self.base_url()}/test_website/308/name-{rec_published.id}")
 
-            resp = self.url_open("/test_website/308/name-%s" % rec_published.id, allow_redirects=False)
+            resp = self.url_open(f"/test_website/308/name-{rec_published.id}", allow_redirects=False)
             self.assertEqual(resp.status_code, 200)
 
-            resp = self.url_open("/test_website/200/xx-%s" % rec_published.id, allow_redirects=False)
+            resp = self.url_open(f"/test_website/200/xx-{rec_published.id}", allow_redirects=False)
             self.assertEqual(resp.status_code, 308)
-            self.assertEqual(resp.headers.get('Location'), self.base_url + "/test_website/308/xx-%s" % rec_published.id)
+            self.assertEqual(resp.headers.get('Location'), f"{self.base_url()}/test_website/308/xx-{rec_published.id}")
 
-            resp = self.url_open("/test_website/308/xx-%s" % rec_published.id, allow_redirects=False)
+            resp = self.url_open(f"/test_website/308/xx-{rec_published.id}", allow_redirects=False)
             self.assertEqual(resp.status_code, 301)
-            self.assertEqual(resp.headers.get('Location'), self.base_url + "/test_website/308/name-%s" % rec_published.id)
+            self.assertEqual(resp.headers.get('Location'), f"{self.base_url()}/test_website/308/name-{rec_published.id}")
 
-            resp = self.url_open("/test_website/200/xx-%s" % rec_published.id, allow_redirects=True)
+            resp = self.url_open(f"/test_website/200/xx-{rec_published.id}", allow_redirects=True)
             self.assertEqual(resp.status_code, 200)
-            self.assertEqual(resp.url, self.base_url + "/test_website/308/name-%s" % rec_published.id)
+            self.assertEqual(resp.url, f"{self.base_url()}/test_website/308/name-{rec_published.id}")
```

Comment:
Ok so let me explain it...

I should had done it in another PR, merged prior to httpocalypse. Now it is too late, it is here and it would be a pain to isolate the changes... Sorry about it :(

Because I'm changing all `self.base_url + ...` to `self.base_url() + `, I change most of the file. Because I change most of the time I took the opportunity to f-string everything (that's the recommended way post py3.6).

There are no interesting changes to review, this is a pure style changed motivated by the changes in the TransactionCase base class. There was duplicated code that we refactored, during the refactor we used the *existing* `base_url()` method much more but the result was that all sub-classes were `base_url` was redefined crashed.

100% I should had done it in another PR. Now it is too late because the `base_url()` changes are coupled with the changes we did in the base test class to make it compatible with the new in-db session storage.

My apologies about it :/